stan/math/prim/fun/qr.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef STAN_MATH_PRIM_FUN_QR_HPP
+#define STAN_MATH_PRIM_FUN_QR_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <algorithm>
+#include <tuple>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the fat QR decomposition
+ *
+ * @tparam EigMat type of the matrix
+ * @param m Matrix.
+ * @return A tuple containing (Q,R):
+ * 1. Orthogonal matrix with maximal columns
+ * 2. Upper triangular matrix with maximal rows
+ */
+template <typename EigMat, require_eigen_t<EigMat>* = nullptr>
```

Comment:
[optional] this could be made tighter with 

```suggestion
template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
```

Which forces the signature to only accept Eigen types with dynamic rows and columns